### PR TITLE
Test sg with btn

### PR DESCRIPTION
**Pull request by Hotell**
this PR
- monkey patches bore with our custom `h` for correct processing of attributes
- adds testing styleguides 
- adds `expect` for assertions instead of `chai`

### CODE REVIEW

**Comment by Hotell** [comment]
this is failing because of https://github.com/skatejs/bore/issues/11

File: docs/styleguides/TESTING.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    });
+  });
+
+  // 2. API
+  describe( `API`, () => {
+    // 2.1 color property ( we use [] brackets like in Angular to make clear what is prop and what is event)
+    describe( `[color]`, () => {
+
+      it( `should render no color class on button by default`, () => {
+        return mount( `<bl-button>huhuhu</bl-button>` )
```
Comment (devopsmariocom) [inline docs/styleguides/TESTING.md:43]: can we use some simple helper which we cane safely remove after skatejs/bore#11 is fixed?

File: docs/styleguides/TESTING.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Testing
+
+What should be tested and how:
+
+1. Component registration to `customElements` registry and initial render
+2. API - primarily props. attributes if we support it. ( if we use attributes reflection jsut test string html version for that API)
```
Comment (devopsmariocom) [inline docs/styleguides/TESTING.md]: typo `jsut` ☂️

File: docs/styleguides/TESTING.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  describe( `Custom element`, () => {
+    it( `should be registered`, () => {
+      expect( customElements.get( Button.is ) ).toBe( Button );
+    });
+    it( `should render via JSX IntrinsicElement `, () => {
+      return mount( <bl-button>Hello</bl-button> ).wait( element => {
+        expect( element.node.localName ).toBe( Button.is );
+      });
+    });
+    it( `should render `, () => {
```
Comment (devopsmariocom) [inline docs/styleguides/TESTING.md:30]: extra space or sentence is not finished 🐇

File: wct.conf.js [hunk 4]
```diff
@@ -1,6 +1,18 @@
 const packageJson = require( './package.json' );
 
 const config = {
+  extraScripts: [
```
Comment (devopsmariocom) [inline wct.conf.js]: is this necessary? if so can you please explain in comment 💂

**Review by devopsmariocom — CHANGES_REQUESTED** [review]

File: wct.conf.js (hunk 4 above)
Comment (Hotell) [reply wct.conf.js]: forgot to remove, thx

File: docs/styleguides/TESTING.md (hunk 1 above)
Comment (Hotell) [reply docs/styleguides/TESTING.md:43]: right, good idea. Will do. others PR can benefit from it as well

File: packages/button/Button.test.tsx [hunk 5]
```diff
@@ -1,18 +1,44 @@
...
+import { Button } from './index';
 
 describe( Button.is, () => {
 
-  it( 'should render', () => {
+describe( `Custom element`, () => {
+  it( `should be registered`, () => {
+    expect( customElements.get( Button.is ) ).toBe( Button );
+  });
+  it( `should render via JSX IntrinsicElement `, () => {
```
Comment (hopkins-tk) [inline packages/button/Button.test.tsx]: Space at the end.

File: packages/button/Button.test.tsx [hunk 6]
```diff
@@ -1,18 +1,44 @@
...
+    return mount( <bl-button>Hello</bl-button> ).wait( element => {
+      expect( element.node.localName ).toBe( Button.is );
+    });
+  });
+  it( `should render `, () => {
```
Comment (hopkins-tk) [inline packages/button/Button.test.tsx]: Space at the end.

**Review by hopkins-tk — CHANGES_REQUESTED** [review]

**Review by devopsmariocom — APPROVED** [review]

**Review by davidrus — APPROVED** [review]
"Pome do toho!!!"

**Comment by Hotell** [comment]
can you pls re-review @hopkins-tk ASAP ? thx

**Review by hopkins-tk — APPROVED** [review]

**Pull request merged**